src/main/java/org/example/daos/JobRoleDao.java

```diff
@@ -85,4 +87,57 @@ public JobRoleInfo getJobRoleById(final int id, final Connection c)
         }
         return null;
     }
+    public void deleteJobRole(final int id, final Connection c) throws
+            SQLException, DatabaseConnectionException {
+        String deleteStatement = "DELETE FROM jobRoles WHERE id = ?";
+        PreparedStatement st = c.prepareStatement(deleteStatement);
+        st.setInt(1, id);
+        st.executeUpdate();
+    }
+
+    public int insertRole(final JobRoleRequest jobRoleRequest,
+                          final Connection c) throws SQLException {
+        String insertRoleQuery =
+                "INSERT INTO jobRoles(role_name, location_id, capability_id,"
+                        + " band_id, closing_date, status, description,"
+                        + " responsibilities, job_spec) VALUES "
+                        + "(?, ?, ?, ?, ?, 'open', ?, ?, ?);";
+
+        PreparedStatement preparedStmt = c.prepareStatement(insertRoleQuery,
+                Statement.RETURN_GENERATED_KEYS);
+        preparedStmt.setString(1, jobRoleRequest.getRoleName());
+        preparedStmt.setInt(2, jobRoleRequest.getLocation());
+        preparedStmt.setInt(3, jobRoleRequest.getCapability());
+        preparedStmt.setInt(4, jobRoleRequest.getBand());
+        preparedStmt.setString(5, jobRoleRequest.getClosingDate().toString());
+        preparedStmt.setString(6, jobRoleRequest.getDescription());
+        preparedStmt.setString(7, jobRoleRequest.getResponsibilities());
+        preparedStmt.setString(8, jobRoleRequest.getLink());
+
+        int affectedRows = preparedStmt.executeUpdate();
+
+        if (affectedRows == 0) {
+            throw new SQLException("Creating user failed, no rows affected.");
+        }
+
+        try (ResultSet rs = preparedStmt.getGeneratedKeys()) {
+            if (rs.next()) {
+                return rs.getInt(1);
+            }
+        }
+
+        return -1;
+    }
+
+    public int getMaxId(final Connection c) throws SQLException {
```

Comment:
I assume you're using this to get the id of a newly created role? but the create role method returns the ID so I don't think this method is needed